staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1266,28 +1276,38 @@ func slicesHaveConflicts(
 	// Compare the slices element by element in order
 	// This test will fail if the slices are not sorted
 	for i := range typedLeft {
-		if hasConflicts, err := mergingMapFieldsHaveConflicts(typedLeft[i], typedRight[i], valueType, "", ""); hasConflicts {
+		if hasConflicts, err := mergingMapFieldsHaveConflicts(typedLeft[i], typedRight[i], valueType, "", nil); hasConflicts {
 			return true, err
 		}
 	}
 
 	return false, nil
 }
 
-func sliceOfMapsToMapOfMaps(slice []interface{}, mergeKey string) (map[string]interface{}, error) {
+func mergeKeysValueToString(mergeKeys []string, m map[string]interface{}) (string, error) {
+	mergeKeyValueStrings := make([]string, 0, len(mergeKeys))
+	for _, mergeKey := range mergeKeys {
+		mergeKeyValue, found := m[mergeKey]
+		if !found {
+			return "", mergepatch.ErrNoMergeKey(m, mergeKey)
+		}
+		mergeKeyValueStrings = append(mergeKeyValueStrings, fmt.Sprintf("%s", mergeKeyValue))
```

Comment:
Here just want to use the string representation of that interface and use the joint strings as the key in the map.